security/src/main/java/zone/cogni/lib/security/saml2/Saml2HttpConfigurer.java

```diff
@@ -41,21 +47,53 @@ public class Saml2HttpConfigurer extends SecurityHttpConfigurer<Saml2HttpConfigu
   private final BasicAuthHandler basicAuthHandler;
   private final GlobalProperties globalProperties;
   private final Saml2Properties saml2Properties;
+  private final String contextPath;
 
   @Override
   public void init(HttpSecurity http) throws Exception {
     log.info("Initializing saml2 security");
     RelyingPartyRegistrationResolver relyingPartyRegistrationResolver = new DefaultRelyingPartyRegistrationResolver(relyingPartyRegistrationRepository);
     Saml2MetadataFilter metadataFilter = new Saml2MetadataFilter(relyingPartyRegistrationResolver, new OpenSamlMetadataResolver());
 
+    Saml2LoginConfigurer<HttpSecurity> httpSecuritySaml2LoginConfigurer = http.saml2Login();
+    checkAssertionConsumerServiceUrl(httpSecuritySaml2LoginConfigurer);
     http
-            .saml2Login().and()
             .apply(new LogoutConfigurer(globalProperties.getLogout())).and()
             .addFilterBefore(this::basicAuthFilter, Saml2WebSsoAuthenticationFilter.class)
             .addFilterBefore(metadataFilter, Saml2WebSsoAuthenticationFilter.class)
             .addFilterAfter(this::patchAuthenticationObjectFilter, Saml2WebSsoAuthenticationFilter.class);
   }
 
+  private void checkAssertionConsumerServiceUrl(Saml2LoginConfigurer<HttpSecurity> httpSecuritySaml2LoginConfigurer) {
+    String assertionConsumerServiceUrl = saml2Properties.getAssertionConsumerServiceUrl();
+    if (StringUtils.isBlank(assertionConsumerServiceUrl) || assertionConsumerServiceUrl.endsWith("/{registrationId}")) return;
+
+    RelyingPartyRegistration registrationId = relyingPartyRegistrationRepository.findByRegistrationId(saml2Properties.getRegistrationId());
+    ObjectPostProcessor<Saml2WebSsoAuthenticationFilter> patchSaml2WebSsoAuthenticationFilterProcessor = new ObjectPostProcessor<>() {
+      @Override
+      public <O extends Saml2WebSsoAuthenticationFilter> O postProcess(O filter) {
+        String assertionConsumerServicePath = getAssertionConsumerServicePath();
+        log.info("Will patch assertionConsumerServicePath in Saml2WebSsoAuthenticationFilter with {}", assertionConsumerServicePath);
+        filter.setFilterProcessesUrl(assertionConsumerServicePath);
+        return filter;
+      }
+
+      private String getAssertionConsumerServicePath() {
+        try {
+          String path = new URL(assertionConsumerServiceUrl).getPath();
+          if (!path.startsWith(contextPath)) throw new RuntimeException("Path of AssertionConsumerServiceUrl does not start with " + contextPath);
+          return "/" + StringUtils.removeStart(path, contextPath);
```

Comment:
if I understand correctly AssertionConsumerServiceUrl should be on the same contextPath as the application? both things are static and could be validated on service startup, no?

Reply:
indeed (if both are set ofc)
but they kindof are validated at startup, the ObjectPostProcessor is used during HttpSecurity init/build